e2e-tests/ldap-tls/run

```diff
@@ -0,0 +1,176 @@
+
+	run_mongo 'db.runCommand({connectionStatus:1})' "percona:password@$cluster-rs0.$namespace" "" "" "--authenticationMechanism 'PLAIN' --authenticationDatabase '\$external'"
+
+	roles_size=$(run_mongo 'JSON.stringify(db.runCommand({connectionStatus:1}))' \
+		"percona:password@$cluster-rs0.$namespace" "" "" \
+		"--authenticationMechanism 'PLAIN' --authenticationDatabase '\$external'" \
+		| grep -E -v "Percona Server for MongoDB|connecting to:|Implicit session:|versions do not match|Error saving history file:|bye" \
+		| jq '.authInfo.authenticatedUserRoles | length')
+
+	if [ -z "$roles_size" ]; then
+		echo "Got problem while getting roles size"
+		exit 1
+	fi
+
+	if [ "$roles_size" -eq 0 ]; then
+		echo "No roles found"
+		exit 1
+	fi
+
+	kubectl_bin delete psmdb $cluster
+	kubectl_bin delete pvc --all
+}
+
+test_sharded_openldap() {
+	cluster="some-name-sharded"
+
+	kubectl_bin create secret generic $cluster-mongos --from-literal=mongos.conf="$(yq "$test_dir/conf/mongos.conf" \
+		| yq ".security.ldap.servers=\"openldap.$namespace.svc.cluster.local:1636\"")"
+
+	kubectl_bin create secret generic $cluster-cfg-mongod --from-literal=mongod.conf="$(yq "$test_dir/conf/mongod.conf" \
+		| yq ".security.ldap.servers=\"openldap.$namespace.svc.cluster.local:1636\"")"
+
+	desc "create first PSMDB cluster $cluster"
+
+	apply_cluster "$test_dir/conf/$cluster.yml"
+
+	desc 'check if all 3 Pods started'
+	wait_for_running $cluster-rs0 3
+	wait_for_running $cluster-cfg 3 "false"
+	wait_for_running $cluster-mongos 3
+
+	run_mongos \
+		'db.getSiblingDB("admin").createRole(
+{
+ role: "cn=admin,ou=perconadba,dc=ldap,dc=local",
+ privileges: [],
+ roles : [
+   {
+     "role" : "readAnyDatabase",
+     "db" : "admin"
+   },
+   {
+     "role" : "dbAdminAnyDatabase",
+     "db" : "admin"
+   },
+   {
+     "role" : "clusterMonitor",
+     "db" : "admin"
+   },
+   {
+     "role" : "readWriteAnyDatabase",
+     "db" : "admin"
+   },
+   {
+     "role" : "restore",
+     "db" : "admin"
+   },
+   {
+     "role" : "backup",
+     "db" : "admin"
+   }
+ ]})' \
+		"userAdmin:userAdmin123456@$cluster-mongos.$namespace"
+
+	roles_size=$(run_mongos 'JSON.stringify(db.runCommand({connectionStatus:1}))' \
+		"percona:password@$cluster-mongos.$namespace" "" "" \
+		"--authenticationMechanism 'PLAIN' --authenticationDatabase '\$external'" \
+		| grep -E -v "Percona Server for MongoDB|connecting to:|Implicit session:|versions do not match|Error saving history file:|bye" \
+		| jq '.authInfo.authenticatedUserRoles | length')
+
+	if [ -z "$roles_size" ]; then
+		echo "Got problem while getting roles size"
+		exit 1
+	fi
+
+	if [ "$roles_size" -eq 0 ]; then
+		echo "No roles found"
+		exit 1
+	fi
+
+	kubectl_bin delete psmdb $cluster
+	kubectl_bin delete pvc --all
+}
+
+main() {
+	deploy_cert_manager
+
+	create_infra "$namespace"
+
+	deploy_openldap
+
+	desc 'create secrets and start client'
+	cluster="some-name"
+	kubectl_bin apply \
```

Comment:
@pooknull I can't find when we create secret with certificate for ldap :( Maybe we need to deploy cert-manager for this test
```
Error from server (InternalError): error when creating "STDIN": Internal error occurred: failed calling webhook "webhook.cert-manager.io": failed to call webhook: Post "https://cert-manager-webhook.cert-manager.svc:443/mutate?timeout=10s": service "cert-manager-webhook" not found
Error from server (InternalError): error when creating "STDIN": Internal error occurred: failed calling webhook "webhook.cert-manager.io": failed to call webhook: Post "https://cert-manager-webhook.cert-manager.svc:443/mutate?timeout=10s": service "cert-manager-webhook" not found
```

Reply:
cert-manager is deployed here: https://github.com/percona/percona-server-mongodb-operator/blob/5620d253ad9717a9ef762ddf01d34a30eadc4dc0/e2e-tests/ldap-tls/run#L156

Certificate is here: https://github.com/percona/percona-server-mongodb-operator/blob/5620d253ad9717a9ef762ddf01d34a30eadc4dc0/e2e-tests/ldap-tls/conf/openldap.yaml#L8-L28

Reply:
ok, you need to have  deploy_cert_manager after create_infra because in create_infra we try to clean up old namespaces

Reply:
We have `deploy_cert_manager` before `create_infra` almost in every other test. `create_infra` only deletes our crds and doesn't touch `cert-manager`. If it's really the problem, then we should update all other tests

Reply:
@pooknull it is from create_namespace function
```
⋮   if [[ ${CLEAN_NAMESPACE} == 1 ]] && [[ -z ${skip_clean_namespace} ]]; then
⋮   ⋮   destroy_chaos_mesh
⋮   ⋮   desc 'cleaned up all old namespaces'
⋮   ⋮   kubectl_bin get ns \
⋮   ⋮   ⋮   | egrep -v "^kube-|^default|Terminating|psmdb-operator|openshift|gke-mcs|^NAME" \
⋮   ⋮   ⋮   | awk '{print$1}' \
⋮   ⋮   ⋮   | xargs kubectl delete ns &
⋮   fi
```

Reply:
https://github.com/percona/percona-server-mongodb-operator/pull/1480/commits/f2cc78f090b4953e1e574e662f8e31c102e25bc2